azure-pipelines.yml

```diff
@@ -0,0 +1,60 @@
+trigger:
+  batch: true
+  branches:
+    include:
+    - cccs-main
+
+variables:
+  containerRegistry: uchimera
+  imageRepository: cccs/trino
+  buildTimestamp: $[format('{0:yyyyMMddHHmmss}', pipeline.startTime)]
+  DOCKER_BUILDKIT: 1
+
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+- bash: |
+    BRANCH_NAME=$(echo "$SYSTEM_PULLREQUEST_SOURCEBRANCH $BUILD_SOURCEBRANCH" | sed -r 's/^\s*(refs\/heads\/)?(\S*).*$/\2/' | sed 's/\//_/g')
+    echo "##vso[task.setvariable variable=BRANCH_NAME]$BRANCH_NAME"
+  displayName: Parse Source Control Branch Name
+- bash: $(Build.SourcesDirectory)/mvnw clean install -DskipTests
+  displayName: Maven build
+- bash: |
+    cd $(Build.SourcesDirectory)/core/docker
+    ./build-local.sh
+    if [ $? -ne 0 ]; then
+      echo "Docker build failed"
+      exit 1
+    fi
+
+    cd $(Build.SourcesDirectory)
+    TRINO_VERSION=$(./mvnw --quiet help:evaluate -Dexpression=project.version -DforceStdout)
+
+    DOCKERFILE="FROM trino:${TRINO_VERSION}
+    LABEL cccs.trino.upstream.version=${TRINO_VERSION}"
+
+    mkdir -p /tmp/trino_build
+    echo "${DOCKERFILE}" > /tmp/trino_build/Dockerfile
```

Comment:
Might be a long shot, but if agents are configured to run multiple builds at the same time, there could be a conflict.  So whether you create a local file in the checkout source directory or something like that, or you create a file in /tmp with a distinctive/unique name (like using $$ (-> prosessPid) in the name)

Reply:
Changed to creating a unique dir (using 'mktemp'). It works! 😄